### DISCUSSION TITLE
Enterprise Distribution intallation problem

### DISCUSSION THREAD

**Discussion by llorenzo** [airsdk/Adobe-Runtime-Support#1386] [Q&A]
Hi, anyone has encountered problems with Enterprise Distribution on iOS 15.1 build with AIR (33.1.1.633).
The App is signed with the latest certificate version and it compiles just fine.
But when installing the Enterprisea App I get an alert saying that it can't be installed cause the integrity of the app can't be verified. Well the setting to allow the Enterprise Distributer in the General Settings is not shown.
Any Idea? Has the Enterprise Distribution been tested for this releas build of AIR?
Thanky you,
Martin

**Comment by ajwfrost** [comment]
Hi

Do you have any external frameworks included with your application? There was a code-signing issue with the 33.1.1.633 release where we were also signing the frameworks' dylib files, but then found out that these must be left unsigned... so it may be worth checking with a more recent build of AIR (or you could strip the signatures from those and re-sign the main app, but that's probably a bit convoluted...)

Other than that though, I believe we've had folk who've got enterprise apps being successfully distributed on the latest iOS versions..

thanks

**Reply by llorenzo** [reply]
Hi Andrew,

I just compiled with AIR 33.1.1.686 with no difference in the result. I'm using Native Extensions. But no Framworks like Firebase or so.

**Reply by llorenzo** [reply]
HI Andrew,

I called out for somebody with a working Enterprise Distribution with the current AIR + iOS15 combo. I did not get an answer yet. Is it possible for your team to confirm that it is working fine?

**Reply by ajwfrost** [reply]
Hi @llorenzo  - I know of two companies that are distributing applications internally via the ad-hoc / enterprise distribution model, where we had to support them with some initial code-signing / certificate problems. So I believe it is possible; we can perhaps try to work out if there's a signature/package problem with what you have there..

You mentioned:
> the setting to allow the Enterprise Distributer in the General Settings is not shown

Is that after trying to install the certificate/profile? Does the profile show up in your settings (assuming the devices are managed)?

If you can send us the IPA, we can do some inspecting of it to ensure the signature and certificate, provisioning profile details, entitlements etc are all what we'd expect for an ad-hoc installation?

thanks

**Reply by llorenzo** [reply]
@ajwfrost I'd be grateful for a little help here. How can I send you the Enterprise Link?

**Reply by ajwfrost** [reply]
Hi @llorenzo  - please use https://transfer.harman.com/requests/0TyJ5QVuDMT30MXPkBZB8o
thanks

**Reply by ajwfrost** [reply]
Hi - thanks for that. It does suffer from a bug in the code signature for frameworks that were presumably included via an ANE: we have a patch version of ADT that could help here, or we have a new release with the fix that should be released within the next day or two.
https://transfer.harman.com/message/gbXrnvk8fLTVDk2kIBp6h7 (it should really be put on top of AIR 33.1.1.686 but I think it only really matters if you're building for Android).
thanks

**Reply by llorenzo** [reply]
That sounds terrific! I will wait until the new release and come back to this thread with the results., thanks.

**Reply by llorenzo** [reply]
Hi @ajwfrost I now swapped the adt.jar from the link above in the AIR 33.1.1.686. Unfortunatly there is no diffrence in the behaviour. Still the integrity can't be verifyed.

**Reply by llorenzo** [reply]
@ajwfrost When removing a certain third party framework it works! So i guess I have to look into that direction. Thank you.

**Reply by ajwfrost** [reply]
@llorenzo okay interesting ... so it might be that there is still a slight bug in how we're signing the frameworks; are you able to provide the IPAs with/without this third party framework and we can look into it?
https://transfer.harman.com/requests/Edf855tAu8ZAfQfR47rhuR
thanks

**Comment by llorenzo** [comment]
Can anyone confirm a working Enterprise Distribution on iOS 15? Thanks.

**Reply by Umanoff** [reply]
Briefly, I can confirm that I've got a number of fairly uncomplicated Enterprise apps running under iOS 15. These are made with 33.1.1.633.   
 A few of them have some odd differences from earlier AIR SDK and/or iOS versions in the execution of some code having to do with dynamic text display. I'm still trying to work those out. But basically the apps are working on iOS 15.
